core/src/main/java/io/grpc/ManagedChannelProvider.java

```diff
@@ -87,6 +87,27 @@ public static ManagedChannelProvider provider() {
     return provider;
   }
 
+  private static ClassLoader getCorrectClassLoader() {
+    if (isAndroid()) {
+      // When android:sharedUserId or android:process is used, Android will setup a dummy
+      // ClassLoader for the thread context (http://stackoverflow.com/questions/13407006),
+      // instead of letting users to manually set context class loader, we choose the
+      // correct class loader here.
+      return ManagedChannelProvider.class.getClassLoader();
```

Comment:
Will the dummy class loader be able to read from the META-INF file?

Also, are there any issues with the Class.forName below?  Doesn't that use the current class loader, which in this case would be the dummy one?